### PR TITLE
added user context for sentry

### PR DESCRIPTION
**Pull request by himankbhalla**
@championswimmer  @witty123  Please review and merge.

### CODE REVIEW

File: routers/verifyemail.js [hunk 1]
```diff
@@ -173,6 +187,7 @@ router.get('/key/:key', function (req, res) {
 
         })
         .catch(function (err) {
+	    Raven.captureException(err);
```
Comment (championswimmer) [inline routers/verifyemail.js]: indentation

File: routers/verifyemail.js [hunk 2]
```diff
@@ -113,6 +120,13 @@ router.get('/key/:key', function (req, res) {
             }
 
             if (req.user) {
+		
+		Raven.setContext({
```
Comment (championswimmer) [inline routers/verifyemail.js]: indentation (4 spaces less than required)

File: routers/verifyemail.js [hunk 3]
```diff
@@ -85,7 +92,7 @@ router.post('/', cel.ensureLoggedIn('/login'), makeGaEvent('submit', 'form', 've
             }
         })
         .catch(function (err) {
-
+	    Raven.captureException(err);
```
Comment (championswimmer) [inline routers/verifyemail.js]: indentation

File: routers/setnewpassword.js [hunk 4]
```diff
@@ -113,6 +117,7 @@ router.post('/', makeGaEvent('submit', 'form', 'setnewpassword'), function (req,
     })
     .catch (function (err) {
         // Could not register user
+	 Raven.captureException(err);
```
Comment (championswimmer) [inline routers/setnewpassword.js]: indentation

File: routers/setnewpassword.js [hunk 5]
```diff
@@ -9,13 +9,17 @@ const passutils = require('../utils/password');
...
 const moment =require('moment');
-
+const Raven = require('raven');
 
 router.post('/', makeGaEvent('submit', 'form', 'setnewpassword'), function (req, res) {
   req.body.key = req.body.key.trim();
   req.body.password = req.body.password.trim();
   req.body.passwordagain = req.body.passwordagain.trim();
-  
+  Raven.setContext({
```
Comment (championswimmer) [inline routers/setnewpassword.js]: why do this separately everywhere ?

File: routers/resetpassword.js [hunk 6]
```diff
@@ -10,14 +10,19 @@ const makeGaEvent = require('../utils/ga').makeGaEvent;
...
+const Raven = require('raven');
 
 router.post('/', makeGaEvent('submit', 'form', 'resetpassword'), function (req, res) {
 
   if(req.body.email.trim() === '') {
     req.flash('error', 'Email cannot be empty');
     return res.redirect('/forgotpassword')
   }
-
+  Raven.setContext({
```
Comment (championswimmer) [inline routers/resetpassword.js]: this can be done at a global level middleware that runs before all routes

File: routers/forgotusername.js [hunk 7]
```diff
@@ -7,10 +7,18 @@ const router = require('express').Router();
 const models = require('../db/models').models;
 const makeGaEvent = require('../utils/ga').makeGaEvent;
 const mail = require('../utils/email');
+const Raven = require('raven');
 
 
 router.post('/', makeGaEvent('submit', 'form', 'forgotusername'), function (req, res) {
 
+  Raven.setContext({
```
Comment (championswimmer) [inline routers/forgotusername.js]: this can be done via global middleware too

**Comment by witty123** [comment]
@himankbhalla once the user is logged in you can set usercontext in a global middleware.

**Pull request merged**